packages/shared/lib/services/on-event-scripts.service.ts

```diff
@@ -84,20 +124,88 @@ export const onEventScriptService = {
-                return res;
+                return res.map(dbMapper.from);
             }
             return [];
         });
     },
-    getByConfig: async (configId: number, event: OnEventType): Promise<OnEventScript[]> => {
-        return db.knex.from<OnEventScript>(TABLE).where({ config_id: configId, active: true, event: toDbEvent(event) });
+
+    getByEnvironmentId: async (environmentId: number): Promise<OnEventScript[]> => {
+        const existingScriptsQuery = await db.knex
+            .select<(DBOnEventScript & { provider_config_key: string })[]>(`${TABLE}.*`, '_nango_configs.unique_key as provider_config_key')
+            .from(TABLE)
+            .join('_nango_configs', `${TABLE}.config_id`, '_nango_configs.id')
+            .where({
+                '_nango_configs.environment_id': environmentId,
+                [`${TABLE}.active`]: true
+            });
+        return existingScriptsQuery.map(dbMapper.from);
+    },
+
+    getByConfig: async (configId: number, event: OnEventType): Promise<DBOnEventScript[]> => {
+        return db.knex.from<DBOnEventScript>(TABLE).where({ config_id: configId, active: true, event: eventTypeMapper.toDb(event) });
+    },
+
+    diffChanges: async ({
+        environmentId,
+        onEventScriptsByProvider
+    }: {
+        environmentId: number;
+        onEventScriptsByProvider: OnEventScriptsByProvider[];
+    }): Promise<{
+        added: Omit<OnEventScript, 'id' | 'fileLocation' | 'createdAt' | 'updatedAt'>[];
+        deleted: OnEventScript[];
+        updated: OnEventScript[];
+    }> => {
+        const res: Awaited<ReturnType<typeof onEventScriptService.diffChanges>> = {
+            added: [],
+            deleted: [],
+            updated: []
+        };
+
+        const existingScripts = await onEventScriptService.getByEnvironmentId(environmentId);
+
+        // Create a map of existing scripts for easier lookup
+        const existingMap = new Map(existingScripts.map((script) => [`${script.configId}:${script.name}:${script.event}`, script]));
+
+        for (const provider of onEventScriptsByProvider) {
+            const config = await configService.getProviderConfig(provider.providerConfigKey, environmentId);
+            if (!config || !config.id) continue;
+
+            for (const script of provider.scripts) {
+                const key = `${config.id}:${script.name}:${script.event}`;
+
+                const maybeScript = existingMap.get(key);
+                if (maybeScript) {
+                    // Script already exists - it's an update
+                    res.updated.push(maybeScript);
+
+                    // Remove from map to track deletions
+                    existingMap.delete(key);
+                } else {
+                    // Script doesn't exist - it's new
+                    res.added.push({
+                        configId: config.id,
+                        name: script.name,
+                        version: '0.0.1',
```

Comment:
Out of curiosity:
Did you talk with Bastien about adding on-events-scripts in the templates, managing their versioning, and being able to enable/disable them via the UI?

Reply:
No but that's why I pinged him in a previous PR to mention that there would still be a fair amount of work to support the features you mentioned (pre-built, dasbhoard, etc...) for on-events scripts